chainer/links/connection/multihead_attention.py

```diff
@@ -0,0 +1,196 @@
+import typing as tp  # NOQA
+
+
+from chainer import functions
+from chainer import initializers
+from chainer import link
+from chainer import links
+from chainer import types  # NOQA
+from chainer import variable
+
+
+InputType = tp.Union[variable.Variable, types.NdArray]
+
+
+class MultiHeadAttention(link.Chain):
+
+    """Multi-Head Attention.
+
+    Args:
+        n_head (int): The number of heads.
+        embedding_size (int):
+            The size of input query vectors
+            and projected query, key, and value vectors.
+        self_attention (bool): If ``True``, this becomes self-attention.
+        ksize (int):
+            The size of input key vectors.
+        vsize (int):
+            The size of input value vectors.
+        attention_dropout (float):
+            The dropout ratio applied to attention before softmax.
+        post_dropout (float):
+            The dropout ratio applied to attention after softmax.
+        scaling (float):
+            The scaler value that defaults to :math:`1/\sqrt(n_{head})`.
+        initialW (:ref:`initializer <initializer>`): Initializer to initialize
+            the weight.
+        initial_bias (:ref:`initializer <initializer>`): Initializer to
+            initialize the bias. If ``None``, the bias will be initialized to
+            zero.
+        nobias (bool): Whether to add bias to projected query, key, and value
+            if the instance of this class is used as self-attention.
+        nobias_kv (bool):
+            If ``True``, no bias is added to projected key and value.
+
+    See: `Attention Is All You Need<https://arxiv.org/abs/1706.03762>`_
+
+    """
+
+    def __init__(
+        self,
+        n_head,                 # type: int
+        embedding_size,             # type: int
+        self_attention=False,   # type: tp.Optional[bool]
+        ksize=None,             # type: tp.Optional[int]
+        vsize=None,             # type: tp.Optional[int]
+        attention_dropout=0.0,  # type: tp.Optional[float]
+        post_dropout=0.0,       # type: tp.Optional[float]
```

Comment:
This value is not used as of now but necessary for cuDNN.
ref: https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnSetAttnDescriptor